xDrip/Model/Glucose/GlucoseReading.swift

```diff
@@ -517,4 +525,56 @@ final class GlucoseReading: Object {
             displayDeltaName = displayGlucose.deltaName
         }
     }
+    
+    static func isOptimalConditionToCalibrate() -> Bool {
+        var optimalCalibrationCondition = false
+        let masterReadings = GlucoseReading.allMasterForCurrentSensor
+        if masterReadings.count >= 3 {
+            //We have at least 3 readings
+            let lastReading = masterReadings[masterReadings.count - 1]
+            let middleReading = masterReadings[masterReadings.count - 2]
+            let firstReading = masterReadings[masterReadings.count - 3]
+            
+            if lastReading.calculatedValue != 0 &&
+                middleReading.calculatedValue != 0 &&
+                firstReading.calculatedValue != 0 {
```

Comment:
Don't compare floats with `==` or `!=`, there's a special custom operator - `~` and `!~`.